dbus/methods_test.go

```diff
@@ -130,6 +133,238 @@ func TestStartStopUnit(t *testing.T) {
 	}
 }
 
+// Ensure that ListUnitsByNames works.
+func TestListUnitsByNames(t *testing.T) {
+	target1 := "systemd-journald.service"
+	target2 := "unexisting.service"
+
+	conn := setupConn(t)
+
+	units, err := conn.ListUnitsByNames([]string{target1, target2})
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var unit *UnitStatus
+	for _, u := range units {
+		if u.Name == target1 {
+			unit = &u
+			break
+		}
+	}
+
+	if unit == nil {
+		t.Fatalf("%s unit not found in list", target1)
+	}
+
+	if unit.ActiveState != "active" {
+		t.Fatalf("%s unit should be active but it is %s", target1, unit.ActiveState)
+	}
+
+	unit = nil
+	for _, u := range units {
+		if u.Name == target2 {
+			unit = &u
+			break
+		}
+	}
+
+	if unit == nil {
```

Comment:
Why should it be found in the list? this is counterintuitive the way it's written